pennylane/transforms/defer_measurements.py

```diff
@@ -40,6 +43,14 @@ def defer_measurements(tape):
         that can be controlled as such depends on the set of operations
         supported by the chosen device.
 
+    .. note::
+
+        Devices that inherit `QubitDevice` **must** be initialized with an additional
+        wire for each mid-circuit measurement after which the measured wire is reused
+        or reset for `defer_measurements` to transform the quantum tape correctly.
+        Such devices should also be initialized without custom wire labels for correct
+        behaviour.
```

Comment:
i see that you use `max(wires) + 1`. can this comment be generalized for new devices that don't need wires either? they'd still need a _circuit_ with standard wire labels

Reply:
Yep, totally can. I still have grievances about this behaviour though. I was thinking of just using `qml.map_wires` but I don't like it because if we have a tape with `wires=[1, 0, 3, 2]` (which is totally possible depending on which wires are used first in the queue), they would get mapped to `wires=[0, 1, 2, 3]`, which would be the same labels but permuted, and the user might not realize that the labels have changed.